hydra-node/test/Hydra/Chain/Direct/StateSpec.hs

```diff
@@ -230,6 +237,13 @@ spec = parallel $ do
         UnsupportedLegacyOutput{} -> property True
         _ -> property False
 
+    prop "reject Commits with more than 100 ADA" $
+      forAllCommitWithMoreThan100ADA $ \case
+        FailedToPostTx{failureReason} ->
+          property $
+            failureReason == "cannot commit more than 100 ADA to the head on mainnet."
```

Comment:
I would make it explicit that it is a, somewhat random, hard-coded limit that has been reached. For instance:

```suggestion
            failureReason == "hard-coded limit of 100 ADA for the commit tx on mainnet reached."
```